hazelcast/src/main/java/com/hazelcast/map/impl/operation/steps/engine/StepAwareOperation.java

```diff
@@ -43,4 +47,15 @@ default void applyState(S state) {
     default Step getStartingStep() {
         return null;
     }
+
+    /**
+     * This method is used to inject {@link Backup#afterRun()} to {@link Step} engine.
+     * <p>
+     * Its goal is to call it on completion of offloaded backup operation.
+     *
+     * @param backupOpAfterRun {@link Backup#afterRun()}
+     */
+    default void setBackupOpAfterRun(Consumer backupOpAfterRun) {
```

Comment:
How do the `beforeRun`s get performed? Are we going to have the same?

Reply:
`beforeRun` is executed as is, issue with `afterRun` fixed here is, we were notifying operation caller before offloaded-backup-op finishes now we notify it after offloaded op is finished. This is done by calling `afterRun`
in Step engine.